# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: target/early/initramfs/etc_rc.d_functions-ansi.in.patch
# Copyright (C) 2008 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

--- ./etc/rc.d/functions-ansi.in.orig	2008-02-15 13:43:05.000000000 +0000
+++ ./etc/rc.d/functions-ansi.in	2008-02-15 13:44:12.000000000 +0000
@@ -16,8 +16,8 @@
 #
 
 ECHO='echo -e'
-ESC='\e'	# \e or \033
-BELL='\a'	# \a or 07
+ESC='\033'	# \e or \033
+BELL='\07'	# \a or 07
 
 ART_SUCCESS="  OK "
 ART_FAILURE=" FAIL"
@@ -46,10 +46,9 @@
 	columns=$( if ! stty size -F /dev/tty; then
 		stty size -F /dev/console
 	fi 2> /dev/null | cut -d' ' -f2 )
-	[ -n "$columns" ] || columns=80
 
 	# fill with dots with an "arrow" pointing down at the end.
-	for (( x=1; x<columns; x++ )); do
+	for x in $( seq 1 ${columns:-80} ); do
 		$ECHO -n '.'
 	done
 	$ECHO -n "$MOVE_LEFT_NEXT$ART_NEXT\r"
